yt_dlp/extractor/twitter.py

```diff
@@ -81,28 +86,80 @@ def _search_dimensions_in_video_url(a_format, video_url):
                 'height': int(m.group('height')),
             })
 
-    def _call_api(self, path, video_id, query={}):
-        headers = {
-            'Authorization': 'Bearer AAAAAAAAAAAAAAAAAAAAANRILgAAAAAAnNwIzUejRCOuH5E6I8xnZz4puTs%3D1Zv7ttfk8LF81IUq16cHjhLTvJu4FA33AGWWjCpTnA',
-        }
-        token = self._get_cookies(self._API_BASE).get('ct0')
-        if token:
-            headers['x-csrf-token'] = token.value
-        if not self._GUEST_TOKEN:
-            self._GUEST_TOKEN = self._download_json(
-                self._API_BASE + 'guest/activate.json', video_id,
-                'Downloading guest token', data=b'',
-                headers=headers)['guest_token']
-        headers['x-guest-token'] = self._GUEST_TOKEN
-        try:
-            return self._download_json(
-                self._API_BASE + path, video_id, headers=headers, query=query)
-        except ExtractorError as e:
-            if isinstance(e.cause, compat_HTTPError) and e.cause.code == 403:
-                raise ExtractorError(self._parse_json(
-                    e.cause.read().decode(),
-                    video_id)['errors'][0]['message'], expected=True)
-            raise
+    def is_logged_in(self):
+        return bool(self._get_cookies(self._API_BASE).get('auth_token'))
+
+    def _call_api(self, path, video_id, query={}, graphql=False):
+        cookies = self._get_cookies(self._API_BASE)
+        base_headers = {}
+
+        csrf_cookie = cookies.get('ct0')
+        if csrf_cookie:
+            base_headers['x-csrf-token'] = csrf_cookie.value
```

Comment:
```suggestion
        cookies = self._get_cookies(self._API_BASE)
        headers = {}

        csrf_cookie = cookies.get('ct0')
        if csrf_cookie:
            headers['x-csrf-token'] = csrf_cookie.value
```

Reply:
If we reuse the `headers` dict we need to delete the `x-guest-token` before requesting a new guest-token. That is why I decided to split it up, so we will always have a clean headers dict.